### PR TITLE
Added support for Translator debug section

### PR DESCRIPTION
**Pull request by iGusev**

### CODE REVIEW

**Comment by iGusev** [comment]
@fabpot

File: WebProfilerServiceProvider.php [hunk 1]
```diff
@@ -215,6 +219,20 @@ public function register(Container $app)
...
+        if (isset($app['translator']) && $app['translator'] instanceof Translator) {
+            $app['data_collectors'] = $app->extend('data_collectors', function ($collectors, $app) {
+                $collectors['translation'] = function ($app) {
+                    return new TranslationDataCollector($app['translator']);
+                };
+
+                return $collectors;
+            });
+
+            $translator = new DataCollectorTranslator($app['translator']);
```
Comment (HeahDude) [inline WebProfilerServiceProvider.php]: use `$app->extend()`

File: WebProfilerServiceProvider.php [hunk 2]
```diff
@@ -215,6 +219,20 @@ public function register(Container $app)
             });
         }
 
+        if (isset($app['translator']) && $app['translator'] instanceof Translator) {
```
Comment (HeahDude) [inline WebProfilerServiceProvider.php]: You should not instantiate the service here (testing `instanceof`). I suggest:
```php
if (isset($app['translator']) && class_exists('Symfony\Component\Translation\DataCollector\TranslationDataCollector')) {
```

**Comment by iGusev** [comment]
@HeahDude done!

File: composer.json [hunk 3]
```diff
@@ -23,7 +23,8 @@
         "symfony/css-selector": "^2.8|^3.0",
         "symfony/debug-bundle": "^2.8|^3.0",
         "symfony/security": "^2.8|^3.0",
-        "symfony/security-bundle": "^2.8|^3.0"
+        "symfony/security-bundle": "^2.8|^3.0",
+        "symfony/translation": "^2.8|^3.0"
```
Comment (HeahDude) [inline composer.json]: There is no need for this requirement unless you intend to add some tests :)

**Comment by Spy-Seth** [comment]
👍

File: composer.json (hunk 3 above)
Comment (iGusev) [inline composer.json]: I added it for dev-typehinting. Remove it?

Comment (HeahDude) [inline composer.json]: Loading a component for type hinting is too much IMO, this should be required when you need autoloading in functional/unit tests.

But let's wait for some other opinions :)

**Comment by fabpot** [comment]
Thank you @iGusev.

**Pull request closed without merging**